src/python/pants/rules/core/core_test_model.py

```diff
@@ -0,0 +1,26 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from builtins import str
+
+from pants.build_graph.address import Address
+from pants.util.objects import SubclassesOf, datatype
+
+
+class Status(object):
+  SUCCESS = str('SUCCESS')
+  FAILURE = str('FAILURE')
+
+
+# This would be called TestResult, but then PyTest outputs a warning that it detected TestResult
+# was a test class, but couldn't treat it like a test class, because it has a constructor.
+class GenericTestResult(datatype([
```

Comment:
https://stackoverflow.com/questions/42268401/how-can-i-get-pytest-to-ignore-test-classes-that-dont-subclass-unittest

Reply:
Is there a particular thing you're suggesting here?

I'm not sure how the existing `TestResult` class doesn't fall foul of this, given I can't see any obvious pytest config in the repo...

Reply:
I'm suggesting renaming this to `TestResult` and marking it:
```
__test__ = false
```
...to resolve the comment you have about naming here.